### PR TITLE
Issue120

### PR DESCRIPTION
**Pull request by mstruijs**
Fixes Issue #120

### CODE REVIEW

File: module/Photo/src/Photo/Mapper/WeeklyPhoto.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * Check whether the given photo has been a photo of the week.
+     * 
+     * @param Photo\Model\Photo $photo
+     * @return boolean
+     */
+    public function isPhotoOfTheWeek($photo)
+    {
+        return !is_null($this->getRepository()->findOneBy(array('photo' => $photo)));
+    }
+
```
Comment (jszanto) [inline module/Photo/src/Photo/Mapper/WeeklyPhoto.php]: hasBeenPhotoOfTheWeek($photo) zou deze functie misschien iets beter beschrijven.

Comment (mstruijs) [inline module/Photo/src/Photo/Mapper/WeeklyPhoto.php]: Waarschijnlijk wel, ja.

File: module/Photo/src/Photo/Model/WeeklyPhoto.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @ORM\Column(type="integer")
+     */
+    protected $id;
+
+    /**
+     * The start date of the week the photo is based on.
+     *
+     * @ORM\Column(type="datetime")
```
Comment (kokx) [inline module/Photo/src/Photo/Model/WeeklyPhoto.php]: Doctrine also has a `date` field, which seems much more appropriate here.

File: module/Photo/src/Photo/Service/Photo.php [hunk 3]
```diff
@@ -220,7 +221,74 @@ public function movePhoto($photoId, $albumId)
...
+    /**
+     * Determine which photo is the photo of the week
+     * 
+     * @param \DateTime $begindate
+     * @param \DateTime $enddate
+     * @return \Photo\Model\Photo|null
+     */
+    private function determinePhotoOfTheWeek($begindate, $enddate)
+    {
+        $results = $this->getHitMapper()->getHitsInRange($begindate, $enddate);
```
Comment (kokx) [inline module/Photo/src/Photo/Service/Photo.php]: Is it not better to make the query in here bigger, to include the photo object itself. And also check if it has been photo of the week?

File: module/Photo/src/Photo/Service/Photo.php [hunk 4]
```diff
@@ -220,7 +221,74 @@ public function movePhoto($photoId, $albumId)
...
+    
+    
+    /**
+     * Determine the preference rating of the photo.
+     * 
+     * @param integer $photoId
+     * @param integer $occurences
+     * @return float
+     */
+    private function photoPreference($photo, $occurences)
```
Comment (kokx) [inline module/Photo/src/Photo/Service/Photo.php]: Why is this private? Actually, why are so many methods here private? Unless explicitly needed, the PSR standards state that functions should be protected.

File: module/Photo/src/Photo/Model/WeeklyPhoto.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    protected $week;
+
+    /**
+     * The photo of the week.
+     *
+     * @ORM\ManyToOne(targetEntity="Photo\Model\Photo", inversedBy="week")
+     * @ORM\JoinColumn(name="photo_id", referencedColumnName="id")
+     */
+    protected $photo;
+
```
Comment (jszanto) [inline module/Photo/src/Photo/Model/WeeklyPhoto.php]: hoort dit een many to one relation te zijn?

Comment (kokx) [inline module/Photo/src/Photo/Model/WeeklyPhoto.php]: Lijkt mij eerder een OneToOne te zijn inderdaad.

Comment (jszanto) [inline module/Photo/src/Photo/Model/WeeklyPhoto.php]: Also, de inversedBy hoeft hier niet aangezien een photo geen week veld heeft (en dat ook niet echt nodig heeft)

File: module/Photo/src/Photo/Service/Photo.php (hunk 3 above)
Comment (mstruijs) [inline module/Photo/src/Photo/Service/Photo.php]: I do not really understand what you mean by making the query bigger, do you mean to test more photos than those that have been visited in the specified date range? Photos that have been photo of the week are excluded in line 265.

Comment (jszanto) [inline module/Photo/src/Photo/Service/Photo.php]: I guess you mean something like adding a `WHERE hit.photo NOT IN (SELECT photo FROM WeeklyPhoto)` to immediately exclude previous photos of the week. However I wonder if making queries bigger helps the overal readability of the code.

**Pull request merged**